Exercise_2.js

```diff
@@ -0,0 +1,272 @@
+const find_index = (name) => {
+  findIndexStart = inventors.findIndex(function (name) {
+    return countSpace(name) <= 1;
+  });
+  return findIndexStart;
+}
+
+const newArray=(name)=> {
+  var array = "";
+  startArray = find_index(inventors);
+  endArray = name.length + 1;
+  array = array + inventors.slice(startArray, endArray);
+  return array;
+}
+ var i = 0;
+ while (i< inventors.length){
+  console.log(newArray(inventors[i]))
+  i++;
+ };
+ 
+
+// Find the number divisible by 7.
+result = numbers.find(function (number) {
+  if (number !== 0) {
+    return number % 7 === 0;
+  }
+}
+);
+console.log(result)
+
+// task 1: Given a string, Write a function to reverse each word in the sentence
+const string = "Nguyen Thi Ut Vien";
+
+const reverseSentence = (string) => {
+  const array = string.split(" ");
+  result = array.map(function (characters) {
+    return characters.split("").reverse().join("");
+  });
+  return result.join(" ")
+};
+console.log(reverseSentence(string));
+
+
+// task 2: Write a recursive function to reverse the string.
+const reverse = (string) => {
+  if (string === "") {
+    return "";
+  }
+  else {
+    return reverse(string.slice(1)) + string.charAt(0);
+  }
+};
+console.log(reverse("Nguyen Thi Ut Vien"))
+
+// task 3: Given an array, Write a function to reverse the items in that array.
+const reverseArray = (array) => {
+  for (var i = 0; i < (array.length - 1); i++) {
+    array.splice(i, 0, array.pop());
+  }
+  return array;
+}
+array = [1, 2, 4, 3]
+console.log(reverseArray(array))
+
+// task 4: write a recursive function that reverses an array
+const reverseArr = (array) => {
+  if (array.length == 0) {
+    return "";
+  }
+  var s = array.shift();
+  return reverseArr(array) + s
+};
+array = [1, 3, 2, 4];
+
+console.log(reverseArr(array))
+
+// task 5: write a recursive sum function
+
+const sum = (number) => {
+  if (number > 0) {
+    return number + sum(number - 1);
+  }
+  else {
+    return number
+  }
+}
+
+console.log(sum(5));
+
+
+// task 6: Write a recursive function to know if a number is prime
+const checkPrime = (number) => {
+  if (number <= 2) {
+    return false
+  }
+  if (number % 2 === 0) {
+    return false;
+  }
+  return checkPrime(number)
+}
+console.log(checkPrime(4))
+
+//task 7: Write a class that includes 4 methods: add, update, remove, compare
+
+class User {
+  users = [{ id: 1, name: 'Lan' },
+  { id: 2, name: 'Hoa' }];
+
+  add(user) {
+    users.push({ id: 3, name: "Tha", age: 30 });
+    console.log(users);
+  };
+
+  // update(user) {
+  //   var user = { id: 1, name: 'Lan' };
+  //   this.users.replace(user);
+  //   console.log(users)
+
+  // };
+
+  remove(id) {
+    var user = { id: 1, name: 'Lan' };
+    user.splice(user.indexOf(user), 1)
+    console.log(users)
+  };
+};
+
+const user = new User();
```

Comment:
{
    class User {
        users = [{ id: 1, name: 'name 1' }, { id: 2, name: 'name 2' }];

        add(user) {
            const isExist = this.users.some(item => item.id === user.id);
            if (isExist) return this.users;
            this.users = [...this.users, user];
            return this.users;
        };

        update(user) {
            const index = this.users.findIndex(t => t.id === user.id);
            this.users[index] = user;
            return this.users;
        };

        remove(id) {
            this.users = this.users.filter(item => item.id !== id);
            return this.users;
        };
    };

    const user = new User();

    // console.log(user.add({ id: 3, name: 'name 3' }));
    // console.log(user.update({ id: 2, name: 'update name 2' }));
    // console.log(user.remove(2));
}